python/docs/graph_business_trip/README.md

```diff
@@ -0,0 +1,23 @@
+# Graph Business Trip
+
+## Challenge Summary
+Write a function called business trip
+- Arguments: graph, array of city names
+- Return: the cost of the trip (if it’s possible) or null (if not)
+
+## Approach & Efficiency
+
+#### Approach
+
+Use a helper function to calculate the cost of a trip between a pair of locations.  Returns the cost if the locations are connected, or False otherwise.
+
+Since a trip may involve more than one location, iterate through the list of locations forming pairs in each iteration.  Then call the cost helper function.  If it returns False, then return False, 0.  Otherwise, add the cost to the cost accumulator.  Once all pairs have been evaluated, return True, cost accumulator.
+
+#### Efficiency
+
+Time: O(n)
```

Comment:
The time complexity on this one is a little tricky.

Line 17 loops through every city in the itinerary, so there's a linear dependency based on itinerary length: an O(N) contribution where N is the size of the itinerary.

At each city, a call to direct() checks whether the two input cities are connected. This function loops through all the keys in the graph to find the starting city: O(V) where V is the number of vertices in the graph. This means there's a linear dependency on the size of the graph as well, at each city in the itinerary. This brings up the run time to O(N*V) where N is the size of the input list and V is the number of vertices in the graph.

Lastly, each edge is being checked one by one that the destination exists: O(E) where E is the number of edges per starting location, keeping in mind that this occurs for each city in the itinerary. Similar to the last paragraph, this adds to the tasks occuring at each N (city in the itinerary): O(N*(V+E)).

Tricky time complexity. If there's anything fishy with what I'm typing or if there are questions, feel free to let me know.